### PR TITLE
Cross-Partitioning TS, Part 4: More Tests, Part 1

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Add more tests to cross-partitioning TS; PR #4097 had a few and we manually verified things worked, but that's not sustainable especially if/when we want to push more changes to TS.

**Implementation Description (bullets)**:
- Add tests for the accumulator and 'which partitions do I need to clean' logic
- Wire up targeted sweep integration tests to read 8 at a shot.

**Testing (What was existing testing like?  What have you done to improve it?)**:
Automated testing was somewhat lacking for #4097, this PR adds that.

**Concerns (what feedback would you like?)**: nothing major

**Where should we start reviewing?**: whenever

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchWithPartitionInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public void cleansUpOldRowIfNeeded() {
+        when(batch.lastSweptTimestamp()).thenReturn(SweepQueueUtils.maxTsForFinePartition(19));
+        SweepBatchWithPartitionInfo batchWithPartitionInfo
+                = SweepBatchWithPartitionInfo.of(batch, ImmutableSet.of(19L));
+        assertThat(batchWithPartitionInfo.partitionsForPreviousLastSweptTs(SweepQueueUtils.minTsForFinePartition(8L)))
+                .containsExactlyInAnyOrder(8L, 19L);
+    }
+
+    @Test
+    public void cleansUpOldRowEvenIfItHasReachedItsEnd() { // Needed for legacy compatibility
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchWithPartitionInfoTest.java]: Good call to include this!

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                NO_DEDICATED_ROWS,
+                PROGRESS_TIMESTAMP + 288));
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).satisfies(batch -> {
+            assertThat(batch.writes()).containsExactlyInAnyOrder(WRITE_INFO_1, WRITE_INFO_2);
+            assertThat(batch.dedicatedRows()).isEqualTo(NO_DEDICATED_ROWS);
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(PROGRESS_TIMESTAMP + 288);
+        });
+        assertThat(batchWithPartitionInfo.finePartitions()).isEqualTo(ImmutableSet.of(
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java]: 🎖

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                DEDICATED_ROWS_2,
+                PROGRESS_TIMESTAMP + 288));
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).satisfies(batch -> {
+            assertThat(batch.dedicatedRows()).isEqualTo(DedicatedRows.of(
+                    ImmutableList.of(SWEEPABLE_CELLS_ROW_1, SWEEPABLE_CELLS_ROW_2)));
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(PROGRESS_TIMESTAMP + 288);
+        });
+        assertThat(batchWithPartitionInfo.finePartitions()).isEqualTo(ImmutableSet.of());
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java]: This is actually a bug, we should still delete the rows with the references to the dedicated rows!

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+                newSweepTimestamp));
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).satisfies(batch -> {
+            assertThat(batch.writes()).containsExactlyInAnyOrder(writeInfo1AtNewerVersion);
+            assertThat(batch.dedicatedRows()).isEqualTo(NO_DEDICATED_ROWS);
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(newSweepTimestamp);
+        });
+
+        // Both must still be present here!
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java]: 👍

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java (hunk 3 above)
Comment (jeremyk-91) [reply atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java]: Yep, makes sense. I've modified the accumulator to map the rows to their partitions too.

**Comment by gmaretic** [comment]
[no release notes]

**Comment by jeremyk-91** [comment]
It's there. I think the new docs/changelog check might not be fully wired up properly and/or might not work for PRs opened before it was introduced.

**Pull request merged**